File: src/_035/kotlin/Solution.kt
package _035.kotlin

/**
 * @author relish
 * @since 2018/04/15
 */
class Solution {
    fun searchInsert(nums: IntArray, target: Int): Int {
        var left = 0
        var right = nums.size - 1
        while (left <= right) {
            val mid = (left + right) / 2
            when {
                nums[mid] == target -> return mid
                nums[mid] < target -> left = mid + 1
                else -> right = mid - 1
            }
        }
        return left
    }
}

fun main(args: Array<String>) {
    println(Solution().searchInsert(intArrayOf(1,3,5,6), 5))
}

File: src/_035/Solution.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
    if (nums.length === 1) {
        return nums[0] >= target?0:1
    }
    for (let i = 0; i < nums.length - 1; i++) {
        if (nums[i] === target) {
            return i
        }
        if (nums[i] < target && target <= nums[i + 1]) {
            return i+1
        }
        if (nums[i] > target) {
            return 0
        }
    }
    return nums.length
};
